app/views/govuk_publishing_components/components/_character_count.html.erb

```diff
@@ -24,3 +22,7 @@
     </div>
   <% end %>
 <% end %>
+<%
+  add_gem_component_stylesheet("error-message")
+  add_gem_component_stylesheet("character-count")
```

Comment:
Don't these normally go at the top? What's the reason for having them at the bottom?

Reply:
This example is a little more complex than the feedback component given in the PR description.

The `character-count` component also renders the [textarea component](https://github.com/alphagov/govuk_publishing_components/pull/3379/files#diff-5ca2449113fd07595d5a64fffc87b170dde4f2feb677f8cfb80cb890ab02e844R18)

By including the `character-count` component stylesheet at the top of the page, we would load the stylesheets in the order below:

- character-count
- textarea

The textarea for margin-bottom is then set to 0, from textarea.scss

```CSS
.gem-c-textarea .govuk-textarea {
    margin-bottom: 0;
}
```

Percy highlighted this visual change, which was fixed by moving the `character-count` component stylesheet to the bottom.

This would then apply the CSS below, and margin-bottom is set to as required.

```CSS
.govuk-character-count .govuk-form-group,
.govuk-character-count .govuk-textarea {
    margin-bottom: 5px;
}
```

This was a similar issue for the error-message stylesheet and other components in the PR where other stylesheets are imported.

We normally don't have to worry about this in our applications as the styles for textarea are included in static and always imported first - https://github.com/alphagov/static/blob/main/app/assets/stylesheets/application.scss#L20.

This would be an issue for other apps using the gem that do not use static.

Reply:
Oooh complex! Good explanation, thanks. Is this something that can be fixed (or even a problem that needs solving?) - not here, but maybe worth raising an issue about it?

Reply:
I think this would only be an issue if you are using the gem in an app that does not use static and not using the individual loading of stylesheets feature. I will raise an issue for this :)